pkg/graveler/graveler.go

```diff
@@ -1474,9 +1474,20 @@ func (g *Graveler) Delete(ctx context.Context, repository *RepositoryRecord, bra
 		})
 }
 
+func (g *Graveler) deleteConsumer(ctx context.Context, repository *RepositoryRecord, branch *Branch, cachedMetaRangeID *MetaRangeID, keysChan <-chan Key) error {
+	for key := range keysChan {
+		err := g.deleteUnsafe(ctx, repository, branch, key, cachedMetaRangeID)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
I think we might be better off with something like this:
```suggestion
```suggestion
	var retErr error
	for key := range keysChan {
		err := g.deleteUnsafe(ctx, repository, branch, key, cachedMetaRangeID)
		if err != nil {
			// do something with err to make it a DeleteError if it is one
			retErr = multierror.Append(retErr, err)
		}
	}
	return retErr
```

I think that will solve _some_ of the issues (and note that [multierror flattens Errors](https://github.com/hashicorp/go-multierror/blob/9974e9ec57696378079ecc3accd3d6f29401b3a0/append.go#L22-L25)).  But I'd still be happy if a test could catch the issues I think that we had.